File: anplayer-plugin/src/main/java/com/xeinebiu/anplayer/plugin/media/extractor/extractor/SearchExtractor.kt
package com.xeinebiu.anplayer.plugin.media.extractor.extractor

import com.xeinebiu.anplayer.plugin.media.extractor.model.descriptor.MediaDescriptor

/**
 * Extractor to handle queries
 */
abstract class SearchExtractor : Extractor<MediaDescriptor>() {

    /**
     * Returns current query
     */
    abstract fun getQuery(): String?
}

File: src/lib/extractor/search.extractor.ts
import { Extractor } from './extractor';
import {
    Album,
    AlbumDetail,
    Author,
    AuthorDetail,
    Media,
    MediaDetail,
} from '../model';

/**
 * Extractor for full-text search results.
 *
 * Returned by `AnPlayerExtractor.getSearchExtractor(query)`. Results may
 * include a mix of media items, albums, and authors depending on what the
 * source returns for the given query.
 */
export type SearchExtractor = Extractor<
    Media | Album | Author | MediaDetail | AlbumDetail | AuthorDetail,
    {
        /**
         * Returns the query string that was used to produce these results.
         */
        getQuery(): string;
    }
>;
